test/crawler/keyword/keywords_test.exs

```diff
@@ -5,15 +5,93 @@ defmodule Crawler.Keyword.KeywordsTest do
+      assert Keywords.list_user_keywords_by_filter_params(expected_user_id) == [expected_keyword]
+    end
+  end
+
+  describe "list_user_keywords_by_filter_params/2" do
+    test "given a valid user id and valid keyword name, returns user's keyword list" do
+      %{id: user_id} = insert(:user)
+      insert(:keyword, user_id: user_id, name: "keyword")
+
+      %{id: expected_user_id} = insert(:user)
+      expected_keyword = insert(:keyword, user_id: expected_user_id, name: "phone")
+      insert(:keyword, user_id: expected_user_id, name: "tv")
+
+      assert [%{name: stored_keyword_name}] =
+               Keywords.list_user_keywords_by_filter_params(expected_user_id, %{
+                 "keyword" => expected_keyword.name
+               })
+
+      assert stored_keyword_name == "phone"
+    end
+
+    test "given a part of valid keyword, returns user's keyword list" do
+      %{id: user_id} = insert(:user)
+      insert(:keyword, user_id: user_id, name: "phone")
+      insert(:keyword, user_id: user_id, name: "tv")
+
+      %{id: expected_user_id} = insert(:user)
+      insert(:keyword, user_id: expected_user_id, name: "phone")
+      insert(:keyword, user_id: expected_user_id, name: "tv")
+
+      assert [%{name: stored_keyword_name}] =
+               Keywords.list_user_keywords_by_filter_params(expected_user_id, %{
+                 "keyword" => "hon"
+               })
+
+      assert stored_keyword_name == "phone"
+    end
+
+    test "given an invalid keyword, returns empty keyword list" do
+      %{id: user_id} = insert(:user)
+      insert(:keyword, user_id: user_id, name: "keyword")
+
+      %{id: expected_user_id} = insert(:user)
+      insert(:keyword, user_id: expected_user_id, name: "phone")
+      insert(:keyword, user_id: expected_user_id, name: "tv")
+
+      assert Keywords.list_user_keywords_by_filter_params(expected_user_id, %{
+               "keyword" => "something"
+             }) == []
+    end
+
+    test "given a nil keyword, returns keyword list" do
+      %{id: user_id} = insert(:user)
+      insert(:keyword, user_id: user_id, name: "keyword")
+
+      %{id: expected_user_id} = insert(:user)
+      insert(:keyword, user_id: expected_user_id, name: "phone")
+
+      assert [%{name: stored_keyword_name}] =
+               Keywords.list_user_keywords_by_filter_params(expected_user_id, %{
+                 "keyword" => nil
+               })
+
+      assert stored_keyword_name == "phone"
+    end
+
+    test "given another filter that is not `keyword`, returns keyword list without filter" do
+      %{id: user_id} = insert(:user)
+      insert(:keyword, user_id: user_id, name: "phone")
+
+      %{id: expected_user_id} = insert(:user)
+      insert(:keyword, user_id: expected_user_id, name: "phone")
+
+      assert [%{name: stored_keyword_name}] =
+               Keywords.list_user_keywords_by_filter_params(expected_user_id, %{
+                 "another_filter" => nil
```

Comment:
using a `nil` for the filter value might not be a best solution, how about just set a string for it

```suggestion
                 "another_filter" =>"filter_value"
```

Reply:
@andyduong1920 This test case is actually about we will filter out `nil` value in case somehow it appears.

```elixir
defp compact_params(params) do
    params
    |> Enum.filter(fn {_key, value} -> value !== "" and not is_nil(value) end)
    |> Enum.into(%{})
  end
```

Reply:
Oh, I got your mean. Using value is good tho